framework/console/controllers/MigrateController.php

```diff
@@ -312,11 +312,28 @@ protected function generateMigrationSourceCode($params)
             $table = mb_strtolower($matches[1], Yii::$app->charset);
         }
 
-        foreach ($foreignKeys as $column => $relatedTable) {
+        foreach ($foreignKeys as $column => $foreignKey) {
+            $relatedColumn = $foreignKey['column'];
+            $relatedTable = $foreignKey['table'];
+            if (empty($relatedColumn)) {
+                $relatedColumn = 'id';
+                $relatedTableSchema = $this->db->getTableSchema($relatedTable);
```

Comment:
You need DB here so call `$this->db = Instance::ensure($this->db, Connection::className());`.